OWML.ModHelper.Menus/ModInputCombinationPopup.cs

```diff
@@ -0,0 +1,185 @@
+        }
+
+        private void AddKeySign(KeyCode key)
+        {
+            Layout.AddPictureAt(_inputHandler.Textures.KeyTexture(key), Layout.ChildCount, ModInputLibrary.ScaleDown);
+        }
+
+        private void UpdateContents()
+        {
+            Layout.Clear();
+            for (var j = 0; j < _combination.Count; j++)
+            {
+                AddKeySign(_combination[j]);
+                if (j < _combination.Count - 1)
+                {
+                    Layout.AddText("+");
+                }
+            }
+            Layout.UpdateState();
+        }
+
+        public void EnableMenu(bool value, string currentCombination)
+        {
+            if (value)
+            {
+                _combination.Clear();
+                foreach (var key in currentCombination.Split('+').Where(key => key != ""))
+                {
+                    _combination.Add(ModInputLibrary.StringToKeyCode(key));
+                }
+            }
+            if (value && !_initialized)
+            {
+                InitializeMenu();
+            }
+            base.EnableMenu(value);
+            UpdateContents();
+        }
+
+        public override void EnableMenu(bool value)
+        {
+            this.EnableMenu(value, "");
+        }
+
+        public void SetUpPopup(string message, SingleAxisCommand okCommand, SingleAxisCommand cancelCommand,
+            SingleAxisCommand resetCommand, ScreenPrompt okPrompt, ScreenPrompt cancelPrompt,
+            ScreenPrompt resetPrompt, bool closeMenuOnOk = true, bool setCancelButtonActive = true)
+        {
+            SetUpPopupCommandsShort(resetCommand, resetPrompt);
+            base.SetUpPopup(message, okCommand, cancelCommand, okPrompt, cancelPrompt, closeMenuOnOk, setCancelButtonActive);
+        }
+
+        private void SetUpPopupCommandsShort(SingleAxisCommand resetCommand, ScreenPrompt resetPrompt)
+        {
+            this._resetCommand = resetCommand;
+            this._resetButton.SetPrompt(resetPrompt, InputMode.Menu);
+        }
+
+        public virtual void SetUpPopupCommands(SingleAxisCommand okCommand, SingleAxisCommand cancelCommand,
+            SingleAxisCommand resetCommand, ScreenPrompt okPrompt, ScreenPrompt cancelPrompt, ScreenPrompt resetPrompt)
+        {
+            SetUpPopupCommandsShort(resetCommand, resetPrompt);
+            base.SetUpPopupCommands(okCommand, cancelCommand, okPrompt, cancelPrompt);
+        }
+
+        public void Initialize(PopupMenu oldPopupMenu, Selectable defaultSelectable, SubmitAction resetAction,
+            ButtonWithHotkeyImageElement resetButton, ILayoutManager layout, IModInputHandler inputHandler)
+        {
+            _inputHandler = inputHandler;
+            var fields = typeof(PopupMenu).GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+            Array.ForEach<FieldInfo>(fields, field => field.SetValue(this, field.GetValue(oldPopupMenu)));
```

Comment:
var fields = ....ToList();
fields.ForEach(field => field.SetValue(this, field.GetValue(oldPopupMenu)));

Reply:
K

Reply:
Still don't really get why you think that List's ForEach is more readable than array's

Reply:
I don't get why anyone would prefer `SomeStaticMethod(object)` over `object.SomeMethod()` :/

Reply:
Because you don't need to perform conversion to use it?